src/test/java/control/TimelineControllerTest.java

```diff
@@ -0,0 +1,77 @@
+package control;
+
+import data.CameraShot;
+import data.ScriptingProject;
+import gui.CameraShotBlock;
+import gui.CameraShotBlockUpdatedEvent;
+import gui.RootPane;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author alex
+ */
+public class TimelineControllerTest {
+    TimelineController timelineController;
+    private ScriptingProject project;
+    private CameraShot shot;
+    private CameraShotBlock shotBlock;
+
+    @Before
+    public void initialize() {
+        RootPane rootPane = Mockito.mock(RootPane.class);
+        timelineController = new TimelineController(rootPane);
+        project = timelineController.getScriptingProject();
+
+        shot = new CameraShot("Shot test", "", 1, 2);
+        project.getCameraTimelines().get(0).addShot(shot);
+
+        shotBlock = Mockito.mock(CameraShotBlock.class);
+        when(shotBlock.getShotId()).thenReturn(shot.getInstance());
+    }
+
+    @Test
+    public void shotChangedLengthTest() {
+        when(shotBlock.getBeginCount()).thenReturn(1.0);
+        when(shotBlock.getEndCount()).thenReturn(5.0);
+        when(shotBlock.getTimetableNumber()).thenReturn(0);
+
+        timelineController.shotChangedHandler(new CameraShotBlockUpdatedEvent(
+                shotBlock, 0
+        ));
```

Comment:
I would keep the `CameraShotBlockUpdatedEvent` paired with it's constructor variables, they belong together. It would make more sense to me to break the line like this:
```
timelineController.shotChangedHandler(
    new CameraShotBlockUpdatedEvent(shotBlock, 0)
);
```